assets/js/main.js

```diff
@@ -0,0 +1,241 @@
+  var GRID_WIDTH = 50;
+  var GRID_HEIGHT = 40;
+
+  // initial game settings
+  // how fast the game update loop runs
+  var START_TICK = 10;
+  // how many update calls per drop move / collision
+  var START_DROPS_SPEED = 10;
+  // how many update calls per pad movement
+  var START_MOVE_SPEED = 5;
+  // how widet he pad is to start
+  var START_PAD_WIDTH = 4;
+  // how many update calls per createDrop
+  var START_DROP_CREATE_TIME = 99;
+
+  // the jquery game element
+  var $game = null;
+  // the pad that catches the drops
+  var $pad = null;
+  // the jquery score element
+  var $score = null;
+  // all the jquery drop elements
+  var $drops = [];
+
+  // all the game state
+  var game = null;
+
+  // the sound of catching
+  var catchSound = null;
+
+  // ================== view helpers ==================
+  // these are all helpers to get and set model info from a jquery element
+  // the conversion to grid is also handled here
+
+  var pxToNum = function (px) {
+    return parseInt(px.replace('px', ''), 10);
+  };
+
+  var getXY = function ($elem) {
+    return {
+      x: pxToNum($elem.css('left')) / GRID_SIZE,
+      y: pxToNum($elem.css('top')) / GRID_SIZE
+    };
+  };
+
+  var move = function ($elem, x, y) {
+    $elem.css({top: y * GRID_SIZE, left: x * GRID_SIZE});
+  };
+
+  var size = function ($elem, width, height) {
+    $elem.css({width: width * GRID_SIZE, height: height * GRID_SIZE});
+  };
+
+  // ================== drops ==================
+  // all functions related to drops
+
+  // create a drop at the top of the screen with random x
+  var createDrop = function () {
+    var $drop = $('<div class="drop"></div>');
+    var x = Math.floor(Math.random() * GRID_WIDTH);
+    var y = 0;
+    move($drop, x, y);
+    size($drop, 1, 1);
+    $game.append($drop);
+    $drops.push($drop);
+    return $drop;
+  };
+
+  // remove the given drop
+  var removeDrop = function ($drop) {
+    $drops.splice($drops.indexOf($drop), 1);
+    $drop.remove();
+  };
+
+  // move all the drops
+  var moveDrops = function () {
+    for (var i = 0; i < $drops.length; i++) {
+      var $drop = $drops[i];
+      var pos = getXY($drop);
+      // move only by one to make collision detection easier
+      move($drop, pos.x, pos.y + 1);
+    }
+  };
+
+  // check if the drops are colliding with pad or bottom of screen
+  var colideDrops = function () {
+    var padPos = getXY($pad);
+    var $missed = [];
+
+    for (var i = 0; i < $drops.length; i++) {
+      var $drop = $drops[i];
+      var pos = getXY($drop);
+      // the drop hit the bottom :(
+      if (pos.y >= GRID_HEIGHT) {
+        $missed.push($drop);
+      } else if (
+        // colliding with the pad
+        pos.x >= padPos.x && pos.x <= padPos.x + game.padWidth &&
+        // out of bounds
+        pos.y === padPos.y) {
+        onDropCatch($drop);
+      }
+    }
+
+    // drop removeal is separated to avoid removing from an array midway
+    // through iterating
```

Comment:
Couple other ways to handle this, copy the array, adjust i while iterating.  This seems the simplest thoughts?